app/components/score_component.html.haml

```diff
@@ -0,0 +1,6 @@
+.score
+  %p.score__text
+    = score_text
+  = render StarsComponent.new(highlight_stars: highlight_stars?, score: score_number) if scored?
```

Comment:
Can I ask how `show_decimal_score` gets passed in to the `StarsComponent`? I can see that it works once it has been passed in, but can't see from the html.haml file how it overwrites the `show_decimal_score: false` in the component initialization. I may be missing something obvious!

Reply:
So I suppose it gets passed in indirectly by virtue of the `highlight_stars?` method in `ScoreComponent`.  The `StarsComponent` doesn't actually display the score itself, just stars, so it's interface doesn't care about showing the score as a decimal directly.  The `StarsComponent` only really cares about whether it should highlight the stars or not,  so that is what's reflected in its interface.